internal/server/server.go

```diff
@@ -178,6 +178,15 @@ func (s *Server) Run(ctx context.Context) error {
 
 	err := group.Wait()
 	s.tel.Close()
+
+	if sqlDB, err := s.db.DB(); err != nil {
+		logging.L.Warn().Err(err).Msg("failed to get database conn to close")
+	} else {
+		if err := sqlDB.Close(); err != nil {
+			logging.L.Warn().Err(err).Msg("failed to close database connection")
+		}
```

Comment:
this might cause the server to hang. Looks like some of the docs around this say:
```
// Close closes the database and prevents new queries from starting.
// Close then waits for all queries that have started processing on the server
// to finish.
```
we might want a timeout around this so we can ensure the app closes in a reasonable amount of time.

Reply:
Waiting for queries to finish seems like a nice behaviour. We should have timeouts on all queries since we set a deadline on the transaction, right? I'd expect queries to have a pretty short timeout.

I don't think we can set a timeout here because there's no `context.Context` being passed in. In the worst case, kubernetes is going to have its own "graceful shutdown timeout", and if the pod hands after SIGTERM, kubernetes should SIGKILL it, which gives us basically the preivous behaviour of no graceful shutdown.

Is there another documented way of setting a timeout to `Close`?

Reply:
Oh, I forgot we don't even have signal handlers yet, so this code will never be reached in production, it's only used in tests.

When we have signal handlers I think we'll want to make sure a second `SIGTERM` or `SIGINT` does a forced shutdown to avoid a potential hang anywhere in the app.